modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1547 @@
+            "    \"location\": \"ENDPOINT/008bba85-451d-414b-87de-c03b5a1f4217\",\n" +
+            "    \"lastModified\": \"2018-08-17T10:34:29Z\",\n" +
+            "    \"resourceType\": \"User\"\n" +
+            "},\n" +
+            "  \"Operations\": [\n" +
+            "    {\n" +
+            "      \"op\": \"replace\",\n" + //REPLACE Operation
+            "      \"value\": {\n" +
+            "        \"nickName\": \"shaggy\"\n" +
+            "      }\n" +
+            "    }\n" +
+            "  ],   \n" +
+            " \"name\": {\n" +
+            "    \"givenName\": \"Kim\",\n" +
+            "    \"familyName\": \"Berry\",\n" +
+            "    \"formatted\": \"Kim Berry\"\n" +
+            "  },\n" +
+            "  \"userName\": \"kimjohn  \",\n" +
+            "  \"password\": \"kim123\",\n" +
+            "  \"id\": \"123\",\n" +
+            "  \"emails\": [\n" +
+            "      {\n" +
+            "        \"type\": \"home\",\n" +
+            "        \"value\": \"johnnew@gmail.com\",\n" +
+            "         \"primary\": true\n" +
+            "      }\n" +
+            "  ]\n" +
+            "}";
+
+    private static final String ENDPOINT_ME = "https://localhost:9443/scim2/Me";
+
+    private MeResourceManager meResourceManager;
+    private UserManager userManager;
+
+    @BeforeMethod
+    public void setUp() {
+
+        meResourceManager = new MeResourceManager();
+        userManager = mock(UserManager.class);
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+    }
+
+    private SCIMResponse getEncodeSCIMExceptionObject(AbstractCharonException exception) {
+
+        JSONEncoder encoder = new JSONEncoder();
+        Map<String, String> responseHeaders = new HashMap<>();
+        responseHeaders.put(SCIMConstants.CONTENT_TYPE_HEADER, SCIMConstants.APPLICATION_JSON);
+        return new SCIMResponse(exception.getStatus(), encoder.encodeSCIMException(exception), responseHeaders);
+    }
+
+    private User getNewUser() throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        return decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING, schema, new User());
+    }
+
+    @DataProvider(name = "dataForGetSuccess")
+    public Object[][] dataToGetSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        User user = getNewUser();
+        String id = user.getId();
+        String name = user.getUserName();
+
+        return new Object[][]{
+                {id, name, null, null, ResponseCodeConstants.CODE_OK, user},
+                {id, name, null, "emails", ResponseCodeConstants.CODE_OK, user},
+                {id, name, "username,meta", null, ResponseCodeConstants.CODE_OK, user},
+                {id, name, "username", "emails", ResponseCodeConstants.CODE_OK, user}
```

Comment:
since the response status code will be `ResponseCodeConstants.CODE_OK` for any successful get request to Me endpoint, no need to pass it from the data holder.

Reply:
okay, Fixed in commit a5699e694bc31369fb4a347d4a0a12177e73775d